### DISCUSSION TITLE
support self types in interface

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#5692] [Language Ideas] [answered]
today we can't reference the type that's implementing the interface without using generic
such as
```C#
interface IA
{
   Myclass A;
}
class Myclass : interface
{}
```
doesn't work
https://stackoverflow.com/questions/8202384/how-to-reference-the-implementing-class-from-an-interface

**Comment by fa1ntl1ght** [comment, marked as answer]
#5413
